src/pytest_html/result.py

```diff
@@ -0,0 +1,287 @@
+
+        if href is not None:
+            self.links_html.append(
+                html.a(
+                    extra.get("name"),
+                    class_=extra.get("format_type"),
+                    href=href,
+                    target="_blank",
+                )
+            )
+            self.links_html.append(" ")
+
+    def _format_time(self, report):
+        # parse the report duration into its display version and return
+        # it to the caller
+        duration = getattr(report, "duration", None)
+        if duration is None:
+            return ""
+
+        duration_formatter = getattr(report, "duration_formatter", None)
+        string_duration = str(duration)
+        if duration_formatter is None:
+            if "." in string_duration:
+                split_duration = string_duration.split(".")
+                split_duration[1] = split_duration[1][0:2]
+
+                string_duration = ".".join(split_duration)
+
+            return string_duration
+        else:
+            # support %f, since time.strftime doesn't support it out of the box
+            # keep a precision of 2 for legacy reasons
+            formatted_milliseconds = "00"
+            if "." in string_duration:
+                milliseconds = string_duration.split(".")[1]
+                formatted_milliseconds = milliseconds[0:2]
+
+            duration_formatter = duration_formatter.replace(
+                "%f", formatted_milliseconds
+            )
+            duration_as_gmtime = time.gmtime(report.duration)
+            return time.strftime(duration_formatter, duration_as_gmtime)
+
+    def _populate_html_log_div(self, log, report):
+        if report.longrepr:
+            # longreprtext is only filled out on failure by pytest
+            #    otherwise will be None.
+            #  Use full_text if longreprtext is None-ish
+            #   we added full_text elsewhere in this file.
+            text = report.longreprtext or report.full_text
+            for line in text.splitlines():
+                separator = line.startswith("_ " * 10)
+                if separator:
+                    log.append(line[:80])
+                else:
+                    exception = line.startswith("E   ")
+                    if exception:
+                        log.append(html.span(raw(escape(line)), class_="error"))
+                    else:
+                        log.append(raw(escape(line)))
+                log.append(html.br())
+
+        for section in report.sections:
+            header, content = map(escape, section)
+            log.append(f" {header:-^80} ")
+            log.append(html.br())
+
+            if ansi_support():
+                converter = ansi_support().Ansi2HTMLConverter(
+                    inline=False, escaped=False
+                )
+                content = converter.convert(content, full=False)
+            else:
+                content = _remove_ansi_escape_sequences(content)
```

Comment:
this block reads like something to go to the ansi module itself - it also looks to melike the import will always try, the module itself can change